### PR TITLE
populate user profile with real data from hasura

### PR DESCRIPTION
**Pull request by iamnotreddy**
- new gql query and types in fetchUserProfile.ts to fetch profile metadata from Hasura
- add users auth object to profile page, feed userID into query parameters to get current user's metadata
- add userProfileMetadata state var to store profile metadata upon useQuery completion
- populate name, email, profile image, joined date onto page

### CODE REVIEW

**Comment by iamnotreddy** [comment]
Screenshot of UI Changes: 

Screen Shot 2022-06-23 at 10 10 31 AM

File: graphql/fetchUserProfile.ts [hunk 1]
```diff
@@ -30,3 +30,28 @@ export const FETCH_USER_PROFILE = gql`
     }
   }
 `;
+
+export const FETCH_USER_PROFILE_METADATA = gql`
```
Comment (vithushan19) [inline graphql/fetchUserProfile.ts:34]: @jasonrajasegaram you might want to work off this query to fetch user badge data for the user profile.

File: pages/profile.tsx [hunk 2]
```diff
@@ -1,103 +1,193 @@
...
 export default function Profile(props) {
+  const { user } = useAuth();
+
+  const { loading, error, data } = useQuery<FetchUserProfileMetadataResponse>(
+    FETCH_USER_PROFILE_METADATA,
+    {
+      variables: {
+        userId: user.uid,
+      },
+      onCompleted: (data) => {
```
Comment (vithushan19) [inline pages/profile.tsx:19]: great work, fetching data from the API looks very clean

File: pages/profile.tsx [hunk 3]
```diff
@@ -1,103 +1,193 @@
...
+  FetchUserProfileMetadataResponse,
+  FETCH_USER_PROFILE_METADATA,
+  UserProfileMetadata,
+} from "../graphql/fetchUserProfile";
+import { useAuth } from "../lib/authContext";
+
+// query to add User's name, username, email, created at date (joined), profile picture, achievements / badges?? prob existing quiz
 
 export default function Profile(props) {
+  const { user } = useAuth();
```
Comment (vithushan19) [inline pages/profile.tsx:11]: what happens if you try to visit this page as a logged out user? 

You can protect this page from logged out users by adding this line to the bottom of the file: 

`Profile.auth = true;`

You can see how I protected the student portal page here: https://github.com/skillify-ca/skillify-web/blob/main/pages/studentPortal/%5Bslug%5D.tsx#L113

**Review by vithushan19 — COMMENTED** [review]
let me know when you need another review. Great work! Reading from a database is light work for you.

just some minor comments.

**Comment by vithushan19** [comment]
use this library to format your date time: https://date-fns.org/

File: pages/profile.tsx [hunk 4]
```diff
@@ -1,103 +1,191 @@
...
+    FETCH_USER_PROFILE_METADATA,
+    {
+      variables: {
+        userId: user.uid,
+      },
+      onCompleted: (data) => {
+        console.log("oncomplete useQuery", JSON.stringify(data.users[0]));
+        setUserProfileMetadata({
+          __typename: data.users[0].__typename,
+          created_at: data.users[0].created_at,
```
Comment (vithushan19) [inline pages/profile.tsx:23]: This should become createdAt: data.users[0].created_at

**Review by vithushan19 — APPROVED** [review]

**Pull request merged**